src/consensus.rs

```diff
@@ -89,6 +107,17 @@ impl<T: Proposition> Consensus<T> {
         signed_vote: SignedVote<T>,
         gen: Generation,
     ) -> Result<VoteResponse<T>> {
+        if self.contains_new_proposal(&signed_vote) {
```

Comment:
Can this not be a supersedes check against voters old vote? i.e. if the voter sends us a new valid vote and we have already decided, then send the voter the decision.

Reply:
What if we receive a second (bad) vote from a voter that we counted in our SM/SM?

Reply:
We reached termination already, and can't do anything about the bad vote anymore. But we must still tell the node that sent us this that we have termination by sending in our SM vote as proof.

Reply:
Supersedes would report an error in this case, but we don't want to leave the node in the dark and there's nothing we can do about the error because we already reached termination.

Reply:
Right, we're trying to decide if a voter is still sending new votes, so it should be safe to just look at that voter's history in isolation.

A new vote always supersedes the voters old vote, so if you receive a new vote from a voter and the new vote does supersede the old, then we know the voter is still active after we've made a decision, and so we can catch them up.

It shouldn't matter if that new vote happens to contain contradictory proposals from other voters

Reply:
>Supersedes would report an error in this case, but we don't want to leave the node in the dark and there's nothing we can do about the error because we already reached termination.

Missed this msg in my last reply.

The supersedes check should only fail if this voter is faulty or we are receiving an old vote, if the voter is faulty, we're ok with leaving them behind, and if it's an old vote, can safely ignore it since it's a replayed packet